### PR TITLE
Nixify

### PR DESCRIPTION
**Pull request by rail**
This adds Nix expressions to nixify shipitsriptworker. It is also
possible to generate a docker image (mostly used as an example, because
there is no config file, nor a way to pass it and run shipitscript) and
a tarball. The generated tarball can be used as a interim solution in
puppet.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -66,3 +66,6 @@ server_config.json
 work_dir
 artifact_dir
 build-tools
+
+#nix
```
Comment (garbas) [inline .gitignore]: i assume this can be removed now

File: nix/default.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
+{ pkgs ? import <nixpkgs> {} }:
```
Comment (garbas) [inline nix/default.nix]: i think we should also pin `nixpkgs`, as you already did for `pypi2nix`. but nixpkgs is a bit different to pin. you can find instructions here https://nixos.wiki/wiki/How_to_fetch_Nixpkgs_with_an_empty_NIX_PATH

Comment (rail) [reply nix/default.nix]: Good idea!

File: .gitignore (hunk 1 above)
Comment (rail) [reply .gitignore]: This is just to keep `git status` clean, it's not related to `filterSource`.

**Comment by rail** [comment]
@MihaiTabara hey, want to review this and try to run on your machine (or VM)?

**Comment by garbas** [comment]
@rail i wonder how hard would it be to readjust `./please` command to have per project please command. so instead of `./please shell <project-name>` you would just do `./please shell` and that should put you into nix shell like before.

i guess we could create `for-project` command which would call any usual command with a project name. and then we would just need to readjust `./please` bash script a little bit to always call this command and pass any other arguments to it.

File: nix/default.nix [hunk 3]
```diff
@@ -1,6 +1,10 @@
-{ pkgs ? import <nixpkgs> {} }:
-
 let
+  nixpkgsRev = "49996e17ae9daf3c3447fb5e8500dc78b4fde995";
```
Comment (garbas) [inline nix/default.nix]: actually we should still have `pkgs` as an argument and also read sha and rev from json file. we should have a please command to update this json files.

**Comment by rail** [comment]
We tested it on mac and the only thing that didn't work was generating the `update` attribute. It failed with "path too long" OSError.

**Pull request merged**